framework/auth/cas.py

```diff
@@ -0,0 +1,88 @@
+import furl
+import requests
+
+from lxml import etree
+
+
+class CasResponse:
+
+    def __init__(self):
+        self._authenticated = False
+        self._status = None
+        self._user = None
+        self._attributes = {}
+
+    @property
+    def authenticated(self):
+        return self._authenticated
+
+    @authenticated.setter
+    def authenticated(self, value):
+        self._authenticated = value
+
+    @property
+    def status(self):
+        return self._status
+
+    @status.setter
+    def status(self, value):
+        self._status = value
+
+    @property
+    def user(self):
+        return self._user
+
+    @user.setter
+    def user(self, value):
+        self._user = value
+
+    @property
+    def attributes(self):
+        return self._attributes
+
+
+class CasClient:
```

Comment:
It might be useful to subclass `BaseClient`: https://github.com/CenterForOpenScience/osf.io/blob/ff4a7e32e6f718cc5672913730f4b6786a9e9b3f/website/util/client.py#L12